apmproxy/client.go

```diff
@@ -46,13 +46,15 @@ const (
 	defaultDataForwarderTimeout time.Duration = 3 * time.Second
 	defaultReceiverAddr                       = ":8200"
 	defaultAgentBufferSize      int           = 100
+	defaultMaxBatchSize         int           = 50
+	defaultMaxBatchAge          time.Duration = 10 * time.Second
```

Comment:
This seems like a long period of time for the extension. How did you come up with 10 seconds?

Reply:
Nice catch. Considering the nature of the lambda function the duration is definitely too big. My first intention of adding `age` factor to the batch was to account for functions with very low throughput. I got caught up in keeping the batch to be of good size.

Ideally, I think this value should be chosen based on the type of workload. So maybe we can expose configuration options for this(?)

For a good default, we need to balance the batch size and the age. Lambda provides a max time of 15 minutes for a function to run but I think most functions would be < 1 second(?). So, we can keep the default as 2 seconds in aim to flush everything by the second invocation maximum. What is your recommendation here?

Reply:
2 seconds sounds reasonable. We might need to expose a config option for this at some point. Right now, given that the logs collection is in Tech Preview, I'd rather suggest to disable the collection if the `maxBatchAge` becomes an issue for customers.

Reply:
@simitt 
> I'd rather suggest to disable the collection if the maxBatchAge becomes an issue for customers.

FYI, the batching also applies to platform metrics. Is this as per your expectations?